src/Symfony/Component/Security/Http/Firewall/AbstractPreAuthenticatedListener.php

```diff
@@ -54,6 +54,15 @@ public function __construct(TokenStorageInterface $tokenStorage, AuthenticationM
      */
     final public function handle(GetResponseEvent $event)
     {
+        $token = $this->tokenStorage->getToken();
+        if(null !== $token && false === ($token instanceof PreAuthenticatedToken) && $token->isAuthenticated()) {
```

Comment:
Yes, it's what I did and so I changed my code.
Maybe I should make a huge PR to change it my way... just kidding ;)